src/OmniSharp.Roslyn.CSharp/Services/Completion/CompletionService.cs

```diff
@@ -242,6 +242,19 @@ public async Task<CompletionResponse> Handle(CompletionRequest request)
                                     break;
                                 }
 
+                                // If the span we are using is re-using part of the typed text we just need to grab the completion an prefix it
+                                // with the existing text. Such as Onenabled -> OnEnabled, this will re-use On of the typed text
+                                if (typedSpan.Start < change.TextChange.Span.Start && typedSpan.Start < change.TextChange.Span.End && typedSpan.End == change.TextChange.Span.End)
+                                {
+                                    var prefix = typedText.Substring(0, change.TextChange.Span.Start - typedSpan.Start);
+                                    
+                                    (insertText, insertTextFormat) = getAdjustedInsertTextWithPosition(change, position, newOffset: 0);
+
+                                    insertText = prefix + insertText;
```

Comment:
what about a situation where insertText should correct the letter casing of the typed text?

Reply:
Would that be triggered in this case? I don't think it will because that is covered by the if above this one, then then typedSpan is replaced as a whole. In this case the part of the span is already correct so the suggestion only contains the part that needs to be added. At least that is what I saw when trying `one<tab>` vs `One<tab>`.  In the first case `one` gets `OnEnabled() ...` suggestion, but `One` gets `Enabled() ...` suggestion. The span of the first suggestion is identical to the typedSpan, while the span of the second suggestion starts 2 characters after th typedSpan, because you already have `On` in the text.